src/components/AddItemModal/AddItemModal.tsx

```diff
@@ -0,0 +1,250 @@
+  const [updateItem, setUpdateItem] = useState<InventoryItem | string | null>('');
+  const [formData, setFormData] = useState<FormData>({
+    name: '',
+    description: '',
+    type: '',
+    category: '',
+    quantity: 0,
+  });
+  const [errorMessage, setErrorMessage] = useState('');
+  const [nameSearch, setNameSearch] = useState<InventoryItem[]>([]);
+
+  const handleInputChange = (field: string, value: string | number) => {
+    setFormData((prevFormData) => ({
+      ...prevFormData,
+      [field]: value,
+    }))
+  }
+
+  const onChangeHandler = (
+    _event: React.SyntheticEvent | React.FocusEvent<HTMLElement>,
+    value?: InventoryItem | string | null,
+    _reason?: AutocompleteChangeReason,
+    _details?: AutocompleteChangeDetails<InventoryItem>
+  ) => {
+    if (value && typeof (value) === 'object') { //If name already exists, type will be an object
+      setUpdateId(value.id);
+      setFormData({
+        name: value.name,
+        description: value.description,
+        type: value.type,
+        category: value.category,
+        quantity: value.quantity,
+      });
+      setUpdateItem(value);
+    } else if (typeof (value) === 'string') {
+      setFormData((prevFormData) => ({
+        ...prevFormData,
+        name: value,
+      }));
+      setUpdateItem(value);
+    }
+  };
+
+  const resetInputsHandler = () => {
+    setFormData({
+      name: '',
+      description: '',
+      type: '',
+      category: '',
+      quantity: 0
+    })
+    handleAddClose();
+    setUpdateItem(null);
+  }
+
+  const onInputChangeHandler = (_event: React.SyntheticEvent, value: string) => {
+    // This function allows the dropdown menu to appear blank when initially clicking on the textbox. When a user types, it then updates the nameSearch options. nameSearch is then fed in as the possible options in the dropdown options
+    if (value) {
+      const filteredItems = originalData.filter(
+        (item) =>
+          item.name.toLowerCase().includes(value.toLowerCase()) || item.description.toLowerCase().includes(value.toLowerCase())
+      );
+      setNameSearch(filteredItems);
+    } else {
+      setNameSearch([])
+    }
+    setFormData((prevFormData) => ({
+      ...prevFormData,
+      name: value,
+    }));
+    setUpdateItem(value);
+  }
+
+  const createItemHandler = async () => {
+    if (formData.type === '' || formData.name === '' || formData.category === '' || formData.quantity === 0 || formData.description === '') {
+      setErrorMessage('Missing Information')
+    } else {
+      try {
+        const response = await fetch(API, { method: "POST", headers: HEADERS, body: JSON.stringify(formData) });
+        if (!response.ok) {
+          throw new Error(response.statusText);
+        } else {
+          setErrorMessage('');
+          fetchData();
+          handleAddClose();
+          resetInputsHandler();
+        }
+      }
+      catch (error) {
+        console.error('Error posting to database:', error);
```

Comment:
Can we use the snackbar component to make the error visible to the end user? Currently it only shows in console in dev tools.

Reply:
Added